### PR TITLE
Cover Pixel

### PR DESCRIPTION
**Pull request by kamilahjae**
Creates cover pixel functionality to
blend colors with alpha values greater
than zero. Also expands RGB values to
RGBA.

### CODE REVIEW

File: render.h [hunk 1]
```diff
@@ -8,11 +8,17 @@
 
 #define swap16( a, b ) { int16_t t = a; a = b; b = t; }
 
+// turn a division by 255 into something
+// much cheaper to calculate
+// for values between 0 and 65534
+#define div255( num ) ( ( num + ( num >> 8 ) ) >> 8 )
+#define lerp255( background, foreground, a ) ( ( uint8_t )div255( ( foreground * a + background * ( 255-a ) ) ) )
```
Comment (AKimZ) [inline render.h]: lerp?

File: draw.c [hunk 2]
```diff
@@ -8,12 +8,12 @@
 void write_ppm(char *filename, frameBuffer * fb)
 {
     FILE *out = fopen (filename, "w");
-        fprintf(out, "P6 %d %d %d\n", fb->width, fb->height, 255 );
+        fprintf(out, "P8 %d %d %d\n", fb->width, fb->height, 255 );
             fwrite( fb->data, sizeof( pixel )*fb->width*fb->height, 1, out );
             fclose(out);
 }
 
-/* Make the picture, store it, and o home */
```
Comment (AKimZ) [inline draw.c:16]: :stuck_out_tongue_closed_eyes:

File: draw.c [hunk 3]
```diff
@@ -8,12 +8,12 @@
 void write_ppm(char *filename, frameBuffer * fb)
 {
     FILE *out = fopen (filename, "w");
-        fprintf(out, "P6 %d %d %d\n", fb->width, fb->height, 255 );
+        fprintf(out, "P8 %d %d %d\n", fb->width, fb->height, 255 );
```
Comment (AKimZ) [inline draw.c]: Is this where you said that write ppm wasn't accepted 8 bit data?

**Comment by AKimZ** [comment]
I'd be interested in understanding more about how this works. Maybe you can walk me through it when we're less feverish? (We can merge it in if it's stable and compiles though)

**Comment by kamilahjae** [comment]
@AKimZ:  lerp stands for linear interpolation: https://unity3d.com/learn/tutorials/modules/beginner/scripting/linear-interpolation. In the case for the color values, I'm finding the value between an r, g, or b color value and the alpha value. Like interpolating between the values of 5 and 7 by 50% to get 6.

File: draw.c (hunk 3 above)
Comment (kamilahjae) [inline draw.c]: @AKimZ: that was an incorrect assumption on my part, the P6 is a "magic number" to specify the file type  ¯_(ツ)_/¯

**Comment by kamilahjae** [comment]
@AKimZ This is working and up-to-date with master! :boom:  Currently using the coverPixel function on the horizontal line function.

**Pull request merged**